ikarus/finiteelements/mechanics/loads/traction.hh

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#include <dune/fufem/boundarypatch.hh>
+
+#include <ikarus/finiteelements/ferequirements.hh>
+#include <ikarus/utils/defaultfunctions.hh>
+
+namespace Ikarus {
+
+  /**
+   * @brief Traction class represents distributed traction load that can be applied.
+   *
+   * @ingroup mechanics
+   *
+   * @tparam DisplacementBasedElement The type of the displacement-based finite element.
+   * @tparam Traits Type of traits for handling finite elements.
+   */
+  template <typename DisplacementBasedElement, typename Traits>
+  class Traction {
+  public:
+    using FERequirementType       = typename Traits::FERequirementType;
+    using LocalView               = typename Traits::LocalView;
+    using GridView                = typename Traits::GridView;
+    static constexpr int myDim    = Traits::mydim;
+    static constexpr int worldDim = Traits::worlddim;
+
+    /**
+     * @brief Constructor for the Loads class.
+     *
+     * @tparam NeumannBoundaryLoad The type for the Neumann boundary load function.
+     * @param p_neumannBoundary Neumann boundary patch.
+     * @param p_neumannBoundaryLoad Neumann boundary load function.
+     */
+    template <typename NeumannBoundaryLoad>
+    explicit Traction(const BoundaryPatch<GridView>* p_neumannBoundary, NeumannBoundaryLoad p_neumannBoundaryLoad)
+        : neumannBoundary{p_neumannBoundary} {
+      if constexpr (!std::is_same_v<NeumannBoundaryLoad, utils::LoadDefault>)
+        neumannBoundaryLoad = p_neumannBoundaryLoad;
+
+      assert(((not p_neumannBoundary and not neumannBoundaryLoad) or (p_neumannBoundary and neumannBoundaryLoad))
+             && "If you pass a Neumann boundary you should also pass the function for the Neumann load!");
+    }
+
+    /**
+     * @brief Const accessor to the underlying displacement-based finite element (CRTP).
+     *
+     * @return Const reference to the underlying displacement-based finite element.
+     */
+    constexpr const DisplacementBasedElement& dbElement() const {
+      return static_cast<DisplacementBasedElement const&>(*this);
+    }
+
+    /**
+     * @brief Calculates the scalar energy for the given traction load.
+     *
+     * @tparam ScalarType The scalar type for the energy.
+     *
+     * @param par The FERequirementType object.
+     * @param dx Optional displacement vector.
+     * @return The scalar energy.
+     */
+    template <typename ScalarType>
+    auto calculateScalar(const FERequirementType& par,
+                         const std::optional<const Eigen::VectorX<ScalarType>>& dx = std::nullopt) const -> ScalarType {
+      const auto uFunction = dbElement().displacementFunction(par, dx);
+      ScalarType energy    = 0.0;
+      const auto& lambda   = par.getParameter(Ikarus::FEParameter::loadfactor);
+      auto element         = dbElement().localView().element();
+      if (not neumannBoundary and not neumannBoundaryLoad) return energy;
```

Comment:
This should be at the very top of the function and return 0.0. Otherwise you do useless things?!